odxtools/database.py

```diff
@@ -125,13 +128,20 @@ def refresh(self) -> None:
         # let the diaglayers sort out the inherited objects and the
         # short name references
         for dlc in self.diag_layer_containers:
-            dlc._finalize_init(self._odxlinks)
+            dlc._finalize_init(self._odxlinks, self._ecu_shared_datas)
```

Comment:
if you need to add such an argument, make it a `Database` object, i.e., this line should become `dlc._finalize_init(self._odxlinks, self)`. I somehow doubt that this is necessary, though...